src/main/resources/templates/db_scripts/migration_scripts/change_progress_total_data_type.sql

```diff
@@ -1,3 +1,14 @@
 UPDATE ${myuniversity}_${mymodule}.job_executions
 SET jsonb = jsonb_set(jsonb #- '{progress, total}', '{progress, total}', to_jsonb((jsonb -> 'progress' ->> 'total')::int))
-WHERE jsonb -> 'progress' IS NOT NULL;
+WHERE jsonb -> 'progress' IS NOT NULL AND jsonb -> 'progress' <> '{}';
+
+UPDATE ${myuniversity}_${mymodule}.job_executions
+SET jsonb = jsonb_set(jsonb, '{progress}', '{"total": 0, "failed": 0, "exported": 0}'::jsonb)
+WHERE jsonb -> 'progress' IS NULL OR jsonb -> 'progress' = '{}';
```

Comment:
Does the order of these 3 UPDATE statements matter? If the order matter, perhaps this statement should be on top so that it would fill in 0s for the properties, then the last UPDATE  statement below, and lastly the UPDATE statement above to cover all the positive cases, otherwise it would throw errors when certain properties are not present?

Reply:
@mtraneis , the order doesn't matter here. I tested the scripts thoroughly and there weren'nt any errors. 
I created several jobExecutiuon entities with postgres null, posrtgres ={}, postgres ={exported: some_int}, and progres = {total:"123"}. All entities were processed correctly and I got all entities having the postgres with all 3 fields and total as an integer.